### PR TITLE
Docs/Improvement of documentation and library

### PR DESCRIPTION
**Pull request by JulStraus**
This PR follows similar PRs in [`EnergyModelsBase`](https://github.com/EnergyModelsX/EnergyModelsBase.jl/pull/41) and [`EnergyModelsRenewableProducers`](https://github.com/EnergyModelsX/EnergyModelsRenewableProducers.jl/pull/22) and [`EnergyModelsCO2`](https://github.com/EnergyModelsX/EnergyModelsCO2.jl/pull/5). It is essentially an improvement on the documentation with a mathematical description of the individual concepts.

The main aim is to:

1. provide an improved description of the individual areas and transmission modes, specifically regarding the underlying math and requirements for input,
2. restructuring the library to reduce the individual page lengths.

The investment tests are furthermore included in the CI.

### CODE REVIEW

**Comment by JulStraus** [comment]
The tests are currently failing as we have not yet registered the version of EMB which allows for deactivating all tests in [Pull request 39](https://github.com/EnergyModelsX/EnergyModelsBase.jl/pull/39). However, it is still possible to check for typos in the documentation.

File: src/model.jl [hunk 1]
```diff
@@ -108,7 +124,8 @@ Loop through all `TransmissionMode` types and create variables specific to each
 This is done by calling the method [`variables_trans_mode`](@ref) on all modes of each type.
 
 The `TransmissionMode` type representing the widest category will be called first. That is,
-`variables_trans_mode` will be called on a `TransmissionMode` before it is called on `PipeMode`-nodes.
+`variables_trans_mode` will be called on a `TransmissionMode` before it is called on
+`PipeMode`-nodes.
```
Comment (espenfb) [inline src/model.jl:128]:
```suggestion
`PipeMode`.
```

File: src/structures/area.jl [hunk 2]
```diff
@@ -58,12 +56,11 @@ A `GeoAvailability` is required if transmission should be included between indiv
...
 # Fields
-- **`id`** is the name/identifier of the node.\n
-- **`input::Array{<:Resource}`** are the input `Resource`s with conversion value `Real`. \
-The latter are not relevant but included for consistency with other formulations.\n
-- **`output::Array{<:Resource}`** are the generated `Resource`s with conversion value `Real`. \
-The latter are not relevant but included for consistency with other formulations.\n
-
+- **`id`** is the name/identifier of the node.
+- **`input::Array{<:Resource}`** are the input `Resource`s with conversion value `Real`.
+  The latter are not relevant but included for consistency with other formulations.
```
Comment (espenfb) [inline src/structures/area.jl:61]: Is this still true when its an array? Or something that is left from when it was a dict?

File: src/structures/area.jl [hunk 3]
```diff
@@ -58,12 +56,11 @@ A `GeoAvailability` is required if transmission should be included between indiv
...
+- **`output::Array{<:Resource}`** are the generated `Resource`s with conversion value `Real`.
+  The latter are not relevant but included for consistency with other formulations.
```
Comment (espenfb) [inline src/structures/area.jl:63]: same as previous comment

File: src/structures/mode.jl [hunk 4]
```diff
@@ -1,16 +1,20 @@
-""" Declaration of the general type for transmission mode transporting resources between areas."""
+"""
+    abstract type TransmissionMode
+
+Declaration of the general type for transmission mode stransporting resources between areas.
```
Comment (espenfb) [inline src/structures/mode.jl:4]:
```suggestion
Declaration of the general type for transmission modes transporting resources between areas.
```

File: docs/src/area_mode/mode.md [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+  which are in turn calculated as:
+
+  ```math
+  \texttt{trans\_pos}[tm, t] - \texttt{trans\_neg}[tm, t] = 0.5 \times
+  \left(\texttt{trans\_in}[tm, t] + \texttt{trans\_out}[tm, t]\right)
+  ```
+
+  !!! todo "Loss calculations"
+      It looks to me that the loss calculations are not equivalent.
+      We have to change that.
```
[the remark this answers is not in the document]
Comment (JulStraus) [reply docs/src/area_mode/mode.md:196]: I would see that as a separate Issue/PR.

File: docs/src/area_mode/mode.md [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+  EMB.multiple(t_{inv}, t)
+  \end{aligned}
+  ```
+
+  while bidirectional transport utilize again the variables ``\texttt{trans\_pos}[tm, t]`` and ``\texttt{trans\_neg}[tm, t]`` as introduced above:
+
+  ```math
+  \begin{aligned}
+  \texttt{trans\_opex\_var}[tm, t_{inv}] = \sum_{t \in t_{inv}} & opex\_var(tm, t) \times \\ &
+  \left(\texttt{trans\_pos}[tm, t] + \texttt{trans\_neg}[tm, t]\right) \times \\ &
```
[the remark this answers is not in the document]
Comment (JulStraus) [reply docs/src/area_mode/mode.md:238]: We can discuss that in more detail in a separate Issue/PR

File: src/structures/area.jl (hunk 2 above)
Comment (JulStraus) [reply src/structures/area.jl:61]: True. That has to be fixed by. Will do.

File: src/structures/area.jl (hunk 3 above)
Comment (JulStraus) [reply src/structures/area.jl:63]: See above

**Review by JulStraus — COMMENTED** [review]
I will incorporate some of the changes, the rest I would leave for a separate Issue/PR. Changing the behavior on loss calculations would in my opinion correspond to a major release, even if it is not breaking backwards compatibility.

**Review by espenfb — APPROVED** [review]

**Pull request merged**